satpy/readers/aws_l1c.py

```diff
@@ -0,0 +1,122 @@
+# Copyright (c) 2023, 2024 Pytroll Developers
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Reader for the Arctic Weather Satellite (AWS) Sounder level-1c data.
+
+Sample data provided by ESA September 27, 2024.
+"""
+
+import logging
+
+import xarray as xr
+
+from .netcdf_utils import NetCDF4FileHandler
+
+logger = logging.getLogger(__name__)
+
+DATETIME_FORMAT = "%Y-%m-%d %H:%M:%S.%f"
+
+AWS_CHANNEL_NAMES = list(str(i) for i in range(1, 20))
+
+
+class AWSL1CFile(NetCDF4FileHandler):
+    """Class implementing the AWS L1c Filehandler.
+
+    This class implements the ESA Arctic Weather Satellite (AWS) Level-1b
+    NetCDF reader. It is designed to be used through the :class:`~satpy.Scene`
+    class using the :mod:`~satpy.Scene.load` method with the reader
+    ``"aws_l1c_nc"``.
+
+    """
+
+    def __init__(self, filename, filename_info, filetype_info, auto_maskandscale=True):
+        """Initialize the handler."""
+        super().__init__(filename, filename_info, filetype_info,
```

Comment:
Seems like the reader class is not tested.

Reply:
Should be tested now.